### DISCUSSION TITLE
How to not set max_length in aragpt2

### DISCUSSION THREAD

**Discussion by AnwarQarout** [aub-mind/arabert#169] [Q&A]
Is there a way to not set max_length at all for the output text of aragpt2?
I was wondering if there's a way to generate the text with the best possible length, so I want the model to stop whenever the text is good

**Comment by WissamAntoun** [comment]
usually the model keeps generating till it hit the `<|endoftext|>` token, if you are using the pipeline from hugging face you can even choose different tokens to stop the generation on, or till it reaches max_length which can be set to 1024 for the maximum possible length.

The model generates  `<|endoftext|>` when it thinks is good enough and nothing can be generated more. hence if you want you can train the model on a custom dataset so it learns when do you want it to actually stop
